packages/node/src/handlers/call-api.ts

```diff
@@ -1,43 +1,77 @@
 
-function buildOptions(
-  ois: OIS,
-  securitySchemeEnvironmentConfigs: SecuritySchemeEnvironmentConfig[],
-  aggregatedApiCall: AggregatedApiCall
-): adapter.BuildRequestOptions {
-  // Don't submit the reserved parameters to the API
-  const parameters = removeKeys(aggregatedApiCall.parameters || {}, RESERVED_PARAMETERS);
+function addMetadataParameters(aggregatedApiCall: AggregatedApiCall, chain: ChainConfig): ApiCallParameters {
+  const parameters = aggregatedApiCall.parameters;
+  switch (aggregatedApiCall.parameters[ReservedParameterName.RelayMetadata]?.toLocaleLowerCase()) {
+    case 'v1':
+      return {
+        ...parameters,
+        _airnode_airnode_id: aggregatedApiCall.airnodeId,
+        _airnode_client_address: aggregatedApiCall.clientAddress,
+        _airnode_designated_wallet: aggregatedApiCall.designatedWallet,
+        _airnode_endpoint_id: aggregatedApiCall.endpointId,
+        _airnode_requester_index: aggregatedApiCall.requesterIndex,
+        _airnode_request_id: aggregatedApiCall.id,
+        _airnode_chain_type: aggregatedApiCall.chainId,
+        _airnode_chain_id: chain.type,
+        _airnode_airnode_rrp: chain.contracts.AirnodeRrp,
+      };
+    default:
+      return parameters;
+  }
+}
 
-  // Fetch secrets and build a list of security schemes
+function buildSecuritySchemes(
+  ois: OIS,
+  securitySchemeEnvironmentConfigs: SecuritySchemeEnvironmentConfig[]
+): SecurityScheme[] {
   const securitySchemeNames = Object.keys(ois.apiSpecifications.components.securitySchemes);
   const securitySchemes = securitySchemeNames.map((securitySchemeName) => {
-    const securityScheme = ois.apiSpecifications.components.securitySchemes[securitySchemeName];
     const securitySchemeEnvironmentConfig = securitySchemeEnvironmentConfigs.find((s) => s.name === securitySchemeName);
     if (!securitySchemeEnvironmentConfig) {
-      return { ...securityScheme, securitySchemeName, value: '' } as SecurityScheme;
+      return { securitySchemeName, value: '' } as SecurityScheme;
     }
     const value = getEnvValue(securitySchemeEnvironmentConfig.envName) || '';
-    return { ...securityScheme, securitySchemeName, value } as SecurityScheme;
+    return { securitySchemeName, value } as SecurityScheme;
   });
+  return securitySchemes;
+}
+
+function buildOptions(
+  chain: ChainConfig,
+  ois: OIS,
+  securitySchemeEnvironmentConfigs: SecuritySchemeEnvironmentConfig[],
+  aggregatedApiCall: AggregatedApiCall
+): adapter.BuildRequestOptions {
+  // Include airnode metadata based on _relay_metadata version number
+  const parametersWithMetadata = addMetadataParameters(aggregatedApiCall, chain);
+
+  // Don't submit the reserved parameters to the API
+  const sanitizedParameters = removeKeys(parametersWithMetadata || {}, RESERVED_PARAMETERS);
+
+  // Fetch secrets and build a list of security schemes
+  const securitySchemes = buildSecuritySchemes(ois, securitySchemeEnvironmentConfigs);
```

Comment:
I think the naming makes this the most confusing. So just to clarify, here we only create objects with a `name` and `value`. We send these with the OIS and then the `adapter` package matches the new objects up with the relevant security scheme to build the "full" security scheme/parameters.

Maybe the type should be called `SecuritySecrets` or `SecuritySchemeSecrets` instead (and the variables changed accordingly)? What we're building here isn't actually a security scheme. I wrote I lot of the original code and I'm still confusing by it 😄

Reply:
:smile: 

> There are only two hard things in Computer Science: cache invalidation and naming things.
> Phil Karlton

Your description is correct. 
1. in call-api.ts we read the secret from secrets.env and create a key/value object (security scheme name / secret). Then we pass this object plus the ois object to the adapter
2. in authentication.ts we match them by securty scheme name and extract the security scheme from ois object in order to add the auth headers to the request.

I think SecuritySchemeSecrets is a much more descriptive name for this key/value array.

Reply:
@andreogle should we also consider changing the config.json key under `enviroment`?

Reply:
Maybe... What do you think the key should be then? We'll probably want to get Burak involved if you want to change `config.json` keys

Reply:
SecuritySchemeValues maybe? I started wondering about this name change in the config.json when you (accurately) said that the secret was not actually a security scheme but maybe I'm just over complicating things.

Reply:
So in config.json, it would look like this?

```json
{
  "environment": {
    "securitySchemeValues": [
      { ... }
    ]
  }
}
```

I'm not sure... I think let's leave config.json changes out of this PR for now. I think it's fine to change the variable and type names though